routes.go

```diff
@@ -97,19 +99,51 @@ type ClaimsRoutesIn struct {
 }
 
 func BuildClaimsRoutes(in ClaimsRoutesIn) {
-	if in.Router != nil {
+	if in.Router != nil && in.Handler != nil {
 		in.Router.Handle("/claims", in.Handler).Methods("GET")
 	}
 }
 
+// CheckServerRequirements is an fx.Invoke function that does post-configuration verification
+// that we have required servers.  The valid server configurations are:
+//
+//    Both keys and issuer present.  Claims is optional in this case
+//    Neither keys or issuer present.  Claims is required in this case
+//
+// Any other arrangements results in an error.
+func CheckServerRequirements(k KeyRoutesIn, i IssuerRoutesIn, c ClaimsRoutesIn) error {
+	if k.Router != nil && i.Router != nil {
+		// all good ... no need to check anything else
+		return nil
+	}
+
+	if k.Router == nil && i.Router == nil {
+		if c.Router == nil {
+			return errors.New("A claims server is required if no keys or issuer server is configured")
+		}
+	}
```

Comment:
```
if k.Router == nil && i.Router == nil {
    if c.Router ! = nil {
        return  nil  // we can short-circuit here
    }
    return errors.New("A claims server is required if no keys or issuer server is configured")
}
```

Reply:
That's a little too branch-y for my tests.  2 `return`s, instead of 1.

Reply:
Oh hold the phone, that's actually a bug.  Sorry about that ... I'll fix